src/DicomWebClient/CLI/Stow.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Nvidia.Clara.Dicom.DicomWeb.Client.CLI
+{
+    [Command("stow", "Use stow to store DICOM instances to a remote DICOMweb server.")]
+    public class Stow : ConsoleAppBase
+    {
+        private readonly IDicomWebClient _dicomWebClient;
+        private readonly ILogger<Stow> _logger;
+
+        public Stow(IDicomWebClient dicomWebClient, ILogger<Stow> logger)
+        {
+            _dicomWebClient = dicomWebClient ?? throw new ArgumentNullException(nameof(dicomWebClient));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        [Command("store", "Retrieves instances within a study")]
+        public async Task Store(
+            [Option("r", "Uniform Resource Locator (URL) of the DICOMweb service")] string rootUrl,
+            [Option("u", "username for authentication with the DICOMweb service")] string username,
+            [Option("p", "password for authentication with the DICOMweb service")] string password,
+            [Option("i", "DICOM file or directory containing multiple DICOM files")] string input,
+            [Option("o", "Output filename")] string outputFilename = "",
+            [Option("s", "unique study identifier; Study Instance UID")] string studyInstanceUid = "",
+            [Option("t", "Time to wait before the request times out, in minutes.  Default 5 minutes.")] int timeout = 5
+            )
+        {
+            Uri rootUri;
+            ValidateOptions(rootUrl, out rootUri);
+            var files = ScanFiles(input);
+            _logger.LogInformation($"Storing {files.Count} instances...");
+
+            _dicomWebClient.ConfigureServiceUris(rootUri);
+            _dicomWebClient.ConfigureAuthentication(Utils.GenerateFromUsernamePassword(username, password));
+
+            DicomWebResponse<string> response = null;
+            try
+            {
+                response = await _dicomWebClient.Stow.Store(studyInstanceUid, files);
+            }
+            catch (ResponseDecodeException ex)
+            {
+                _logger.LogError($"Error decoding response: {ex.Message}");
+                return;
+            }
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(outputFilename))
+                {
+                    outputFilename = $"{DateTime.Now.ToString("yyyyMMdd-hhmmss-fffff")}-.json";
+                }
+                await Utils.SaveJson(_logger, outputFilename, response.Result);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Error saving results: {ex.Message}");
+            }
+        }
+
+        private IList<DicomFile> ScanFiles(string input)
+        {
+            var attr = File.GetAttributes(input);
+            var dicomFiles = new List<DicomFile>();
+            if (attr.HasFlag(FileAttributes.Directory))
```

Comment:
`enum.HasFlag` is an incredibly expensive way of checking `(attr & FileAttributes.Directory) != 0`

Reply:
Seems like it's ben optimized https://github.com/dotnet/coreclr/pull/13748